src/components/Button/Button.tsx

```diff
@@ -1,55 +1,138 @@
   readonly disabled?: boolean;
+  readonly className?: string;
+  readonly children: React.ReactNode;
+  readonly onClick?: React.MouseEventHandler<HTMLButtonElement>;
 }
 
 export const Button: React.FC<ButtonProps> = ({
+  type = 'button',
+  fontSize,
+  paddingX,
+  paddingY,
+  fontColor,
+  hoverFontColor,
+  bgColor = '#02c1b0',
+  hoverBgColor,
+  hoverBgColorTransitionDuration,
+  borderColor,
+  hoverBorderColor = '#02c1b0',
+  focusBorderColor,
+  borderWidth = 2,
+  borderRadius,
+  minButtonWidth,
+  disabled,
+  className,
   children,
   onClick,
-  bgColor = '#3EDC64',
-  color = 'white',
-  type,
-  className,
-  disabled = false,
   ...baseButtonProps
 }) => {
   return (
     <StyledButton
       type={type}
-      className={className}
+      fontSize={fontSize}
+      paddingX={paddingX}
+      paddingY={paddingY}
+      fontColor={fontColor}
+      hoverFontColor={hoverFontColor}
       bgColor={bgColor}
-      color={color}
-      onClick={onClick}
+      hoverBgColor={hoverBgColor}
+      hoverBgColorTransitionDuration={hoverBgColorTransitionDuration}
+      borderColor={borderColor}
+      hoverBorderColor={hoverBorderColor}
+      focusBorderColor={focusBorderColor}
+      borderWidth={borderWidth}
+      borderRadius={borderRadius}
+      minButtonWidth={minButtonWidth}
       disabled={disabled}
+      className={className}
+      onClick={onClick}
       {...baseButtonProps}>
       {children}
     </StyledButton>
   );
 };
 
 const StyledButton = styled.button<{
+  fontSize?: number;
+  paddingX?: number;
+  paddingY?: number;
+  fontColor?: string;
+  hoverFontColor?: string;
   bgColor: string;
-  color: string;
-  disabled: boolean;
+  hoverBgColor?: string;
+  hoverBgColorTransitionDuration?: number;
+  borderColor?: string;
+  hoverBorderColor: string;
+  focusBorderColor?: string;
+  borderWidth: number;
+  borderRadius?: number;
+  minButtonWidth?: number;
+  disabled?: boolean;
 }>`
-  font-size: 1rem;
-  color: ${({ color }) => color};
+  font-size: ${({ fontSize }) => pxToRem(fontSize ?? 16)};
+  white-space: nowrap;
+  color: ${({ fontColor }) => fontColor ?? `${colors.secondary.White}`};
   background-color: ${({ bgColor }) => bgColor};
-  text-align: center;
-  padding: 0.5rem 1.25rem;
-  border-radius: ${pxToRem(10)};
-  border: none;
-  opacity: ${({ disabled }) => (disabled ? 0.6 : 1)};
+  padding: ${({ paddingX, paddingY }) => {
+    const x = pxToRem(paddingX ?? 16);
+    const y = pxToRem(paddingY ?? 16);
+
+    return `${y} ${x}`;
+  }};
+  border: ${({ borderColor, borderWidth, bgColor }) => {
+    const color = borderColor ?? bgColor;
+    const width = borderWidth ? pxToRem(borderWidth) : '0.125rem';
+    return `solid ${color} ${width}`;
+  }};
+  border-radius: ${({ borderRadius }) => pxToRem(borderRadius ?? 10)};
   cursor: ${({ disabled }) => (disabled ? 'not-allowed' : 'pointer')};
+  min-width: ${({ minButtonWidth }) => pxToRem(minButtonWidth ?? 128)};
```

Comment:
Same with this, I think we should be less opinionated here.  And just set a min-width only if the prop exists.  Otherwise set it to 0 (default out of the box)